pkg/volume/csi/csi_client.go

```diff
@@ -177,13 +179,25 @@ func (c *csiDriverClient) NodeGetInfo(ctx context.Context) (
 	accessibleTopology map[string]string,
 	err error) {
 	klog.V(4).Info(log("calling NodeGetInfo rpc"))
-	if c.nodeV1ClientCreator != nil {
-		return c.nodeGetInfoV1(ctx)
-	} else if c.nodeV0ClientCreator != nil {
-		return c.nodeGetInfoV0(ctx)
-	}
 
-	err = fmt.Errorf("failed to call NodeGetInfo. Both nodeV1ClientCreator and nodeV0ClientCreator are nil")
+	// TODO retries should happen at a lower layer (issue #73371)
+	backoff := wait.Backoff{Factor: 2.0, Steps: 5}
+	err = wait.ExponentialBackoff(backoff, func() (bool, error) {
+		var getNodeInfoError error
+		if c.nodeV1ClientCreator != nil {
+			nodeID, maxVolumePerNode, accessibleTopology, getNodeInfoError = c.nodeGetInfoV1(ctx)
+		} else if c.nodeV0ClientCreator != nil {
+			nodeID, maxVolumePerNode, accessibleTopology, getNodeInfoError = c.nodeGetInfoV0(ctx)
+		}
+		if nodeID != "" {
+			return true, nil
+		}
+		if strings.Contains(getNodeInfoError.Error(), "no handler registered for plugin type") {
+			return false, getNodeInfoError
```

Comment:
Should this be `true` instead of false, meaning condition is satisfied don't retry?

Also nit: comment on what's happening here (e.g. `kubelet plugin registration service not implemented is a terminal error, no need to retry`).

Reply:
I think it can be either true or false here, because the exponential backoff will stop retrying if err is not nil:
```
func ExponentialBackoff(backoff Backoff, condition ConditionFunc) error {
	for backoff.Steps > 0 {
		if ok, err := condition(); err != nil || ok {
			return err
		}
```
 I can change it to true if it makes it more readable